### DISCUSSION TITLE
Are you sure you can use the name "Val"?

### DISCUSSION THREAD

**Discussion by KellanHiggins** [hylo-lang/hylo#774] [Q&A] [answered]
Hi,

Are you aware that Staubli Robotics already has a programming language named Val 3, which is the third iteration of their robot control language?

**Comment by kyouko-taiga** [comment, marked as answer]
Hi,

It is a discussion we might have in the future but we'll stick with Val for the moment.

**Comment by jmafc** [comment]
FWIW, if I'm not mistaken Val 3 is the Variable Assembly Language (`VAL`) described in Wikipedia and it seems like the initial capital spelling, i.e., `Val`, is not used to refer to it.

**Reply by KellanHiggins** [reply]
Fair enough, case does matter in programming!

**Comment by kyouko-taiga** [comment]
Actually, we renamed. Now we're Hylo!

Having an exact homonym might be confusing, even if the case isn't the same.
Besides, the V-space was already quite busy with other programming languages.

**Reply by jmafc** [reply]
OK, I was able to update and rebuild hyloc. I see the executable is still named `valc`. Am I correct in assuming that it will be renamed to `hyloc` at some point (and the source suffix will change as well)? Also, if possible, you may still want to keep the val-lang.dev site around and/or somehow redirect it to hylo-lang.org. I learned about Val from a Phil Nash talk and at first I had trouble finding it because of `V` so having the old site may help others.

**Comment by jmafc** [comment]
@kyouko-taiga JIC you haven't noticed: the tour.hylo-lang.org link on the hylo-lang.org home page goes to the Hylo IR gitbook, rather than whas originally the tour, i.e., the hello-world example and so forth.

**Comment by KellanHiggins** [comment]
Wowee, that was quick since I posted! Good luck. Hylo is definitely easier to find :).

**Comment by Tikkoneus** [comment]
For what it's worth I've been getting very interested in this language and I think Val is by far the better name. It's so simple and elegant and captures the spirit of the language. My $.02.

**Reply by dabrahams** [reply]
We liked "Val" too, but when you receive a cease-and-desist letter from the trademark holder, you have to rethink.

**Reply by Tikkoneus** [reply]
Oh no :( that makes a lot of sense! Thanks for the explanation.
